lib/diagnostics_channel.js

```diff
@@ -23,11 +29,44 @@ const { triggerUncaughtException } = internalBinding('errors');
 
 const { WeakReference } = internalBinding('util');
 
+function decRef(channel) {
+  channel._weak.decRef();
+  if (channel._weak.getRef() === 0) {
+    delete channels[channel.name];
+  }
+}
+
+function markActive(channel) {
+  // eslint-disable-next-line no-use-before-define
+  ObjectSetPrototypeOf(channel, ActiveChannel.prototype);
+  channel._subscribers = [];
```

Comment:
should these be private fields and methods instead of public properties?

Reply:
Ignore the diagnostics_channel parts, that was from a separate PR which needs a rebase now that it has landed. I'll get back to this sometime soon...